### PR TITLE
moved createBackupUntilIncrementalBackup to backup_helper.go

### PR DESCRIPTION
**Pull request by ak-px**
**What this PR does / why we need it**:
moved createBackupUntilIncrementalBackup to backup_helper.go as the methods was not returning proper error message at tests 

**Which issue(s) this PR fixes** (optional)
Closes #PA-1740

**Special notes for your reviewer**:
Tested with the tests which use this method.
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/3061/
https://aetos.pwx.purestorage.com/resultSet/testSetID/385408

### CODE REVIEW

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by snigdha-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -5197,3 +5198,93 @@ func compareNamespaceAge(oldNamespaceAge map[string]time.Time) error {
...
+	if err != nil {
+		return "", fmt.Errorf("creation and validation of incremental backup [%s] creation: error [%v]", incrementalBackupName, err)
+	}
+
+	log.InfoD("Check if backups are incremental backups or not")
+	backupDriver := Inst().Backup
+	ctx, err = backup.GetAdminCtxFromSecret()
+	if err != nil {
+		return "", fmt.Errorf("fetching px-central-admin ctx")
+	}
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:5216]: ?

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -5197,3 +5198,93 @@ func compareNamespaceAge(oldNamespaceAge map[string]time.Time) error {
...
+			// as the original incremental backup which we took has taken a full backup this is mostly
+			// because CloudSnap is taking full backup instead of incremental backup as it's hitting one of
+			// the if else condition in CloudSnap which forces it to take full instead of incremental backup
+			log.InfoD("New backup wasn't an incremental backup hence recreating new backup")
+			listOfVolumes := make(map[string]bool)
+			noFailures := true
+			for maxBackupsBeforeIncremental := 0; maxBackupsBeforeIncremental < 8; maxBackupsBeforeIncremental++ {
+				log.InfoD(fmt.Sprintf("Recreate incremental backup iteration: %d", maxBackupsBeforeIncremental))
+				// Create a new incremental backups
+				incrementalBackupName = fmt.Sprintf("%s-%s-%v", "incremental-backup", namespace, time.Now().Unix())
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:5248]: How about including index in backup name?

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Pull request merged**